source/index.md

```diff
@@ -568,26 +566,76 @@ otherwise, exactly one payment method must be used.
 
 ##### Method: `applepay`
 
-<dl class="dl-horizontal">
-  <dt>applepay[token]</dt>
-  <dd>[:json:] <br /> Full, raw <code>PKPaymentToken</code> object, UTF-8 encoded serialization of a JSON dictionary.</dd>
-  <dt>applepay[symmetric_key]</dt>
-  <dd>[:hex:] <br /> EC symmetric AES key that can decrypt <code>data</code>
-      from the <code>PKPaymentToken</code>; hex formatted as it is a SHA256.</dd>
-</dl>
+Apple Pay requires data of the payment token to be decrypted. For this, the
+merchant's private key is needed to derive a symmetric key which is then used to
+decrypt the PAN, expiry etc.
+
+You can either
+
+* send us the private key and the merchant id (either directly or
+  indirectly by sending the merchant's certificate), then we will derive
+  the symmetric key and decrypt the payment details; or
+
+* derive the symmetric key from the payment token's ephemeral public key
+  together with the merchant's private key and merchant id/certificate.
 
 <p class="alert alert-info">
   <b>Notice:</b> An authorization made with <code>applepay</code> is
   multi-factor authenticated.
   <br /_>
   <b>Notice:</b> An authorization made with <code>applepay</code> may be fully
   3-D Secured, 3-D Secure attempted, or with no 3-D Secure; this is indicated by
-  the <code>eciIndicator</code> of the <code>applepay[token]</code>.
+  the <code>eciIndicator</code> of the <code>applepay[payment_token]</code>.
   <br />
   <b>Notice:</b> An authorization made with <code>applepay</code> cannot be a
   subsequent recurring authorization.
 </p>
 
+###### Using the symmetric key
+
+Your systems must derive the symmetric key. For this, we refer to [our reference
+implementation](https://github.com/clearhaus/pedicel) written in Ruby; see
+[Apple's documentation for the <code>PaymentToken</code>
+object](ApplePay-PaymentToken) for more information.
+
+<dl class="dl-horizontal">
+  <dt>applepay[payment_token]</dt>
+  <dd>[:json:] <br /> Full, raw <code>PKPaymentToken</code> object, UTF-8 encoded serialization of a JSON dictionary.</dd>
+  <dt>applepay[symmetric_key]</dt>
+  <dd>[:hex:] <br /> EC symmetric AES key (unique per transaction) that can
+    decrypt <code>data</code> from the <code>PKPaymentToken</code>; hex
+    formatted as it is a SHA256.</dd>
+</dl>
+
+###### Using the EC private key
+
+When using the EC private key the merchant id is also necessary to decrypt the
+payment token data. This can be extracted from the merchant's certificate.
+
+<dl class="dl-horizontal">
+  <dt>applepay[payment_token]</dt>
+  <dd>[:json:] <br /> Full, raw <code>PKPaymentToken</code> object, UTF-8 encoded serialization of a JSON dictionary.</dd>
+  <dt>applepay[private_key]</dt>
+  <dd>[:PEM:] <br />
+    PEM formatted merchant EC private key. <br />
```

Comment:
Suggestion: s/PEM formatted/PEM-formatted/ (and elsewhere)

http://www.grammar.cl/english/compound-adjectives.htm 🙂

Reply:
Likewise for "Hex encoded".

Reply:
Neither hex nor PEM did indeed format anything 🙂